src/test_l1_dsFPD.c

```diff
@@ -1301,57 +1291,51 @@ void test_l1_dsFPD_negative_dsSetFPColor (void)
     UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
     // Step 03: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
+    enableFPDIndicators(count);
     // Step 04: Call dsSetFPColor() with all valid indicators but invalid colors
     for (int i = 1; i <= count; i++)
     {
         snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
         eIndicator =(dsFPDIndicator_t)UT_KVP_PROFILE_GET_UINT32(buffer);
-        result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
-        UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-	snprintf(supportedColorbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/supportedColors",eIndicator);
+        snprintf(supportedColorbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/supportedColors",i);
         numOfSupportedColors =UT_KVP_PROFILE_GET_LIST_COUNT(supportedColorbuffer);
 
-        for (int j = 0; j < sizeof(allColors)/sizeof(allColors[0]); ++j)
+        for (int j = 0; j < sizeof(dsFPAvailableColors)/sizeof(dsFPAvailableColors[0]); ++j)
         {
-            int isSupported = 0;
-            for (int k = 1; k < numOfSupportedColors; ++k)
+            bool isSupported = false;
+            for (int k = 0; k < numOfSupportedColors; ++k)
             {
-                snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/supportedColors/%d", eIndicator, j);
-                color =(dsFPDColor_t)UT_KVP_PROFILE_GET_UINT32(buffer);
-                if (allColors[j] == color)
+                snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/supportedColors/%d", i, k);
+                color = (dsFPDColor_t)UT_KVP_PROFILE_GET_UINT32(buffer);
+                if(color == dsFPAvailableColors[j])
                 {
-                    isSupported = 1;
+                    isSupported = true;
                     break;
                 }
 
             }
-            if (!isSupported)
-            {
-                result = dsSetFPColor(eIndicator, allColors[j]);
+            if (!isSupported){
+                result = dsSetFPColor(eIndicator, dsFPAvailableColors[j]);
```

Comment:
please check what dsSetFPColor will return if we try to set the unsupported color? 
as per the interface file it is dsERR_OPERATION_NOT_SUPPORTED .

Reply:
updated